app/src/main/java/ch/hikemate/app/utils/PermissionUtils.kt

```diff
@@ -0,0 +1,52 @@
+package ch.hikemate.app.utils
+
+import android.content.Context
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.MultiplePermissionsState
+
+/**
+ * Utility class for handling permissions.
+ *
+ * Strongly inspired (if not copy-pasted) from this Stack Overflow answer:
+ * https://stackoverflow.com/a/40639430
+ */
+object PermissionUtils {
+  /** The name of the shared preferences file that stores persistent data about permissions. */
+  private const val PERMISSIONS_SHARED_PREFERENCES = "permissions"
+
+  /**
+   * Sets whether the user has been asked for a permission before.
+   *
+   * @param context The context to use.
+   * @param permission The permission to set the first time asking status for.
+   * @param isFirstTime Whether the user has been asked for the permission before.
+   * @see firstTimeAskingPermission
+   */
+  fun setFirstTimeAskingPermission(context: Context, permission: String, isFirstTime: Boolean) {
+    val sharedPreferences =
```

Comment:
Sonar: I deem Sonar's suggestion unnecessary since the `sharedPreferences` don't hold sensitive data. So this can stay unencrypted.

Reply:
Maybe you can mark the warning as accepted on SonarCloud? Or should I do this myself?